trace_v2.sh

```diff
@@ -103,175 +92,276 @@ main() {
+	done
+	echo $pid
+}
 
-		# # Save PID/TID map for later reference
-		# docker top $container_name -efT > ${output_dir}/pids_$(date +'%m%d%H%M%S').out
+getPIDOfRootOfContainer() {
+	pid=$(docker inspect -f '{{.State.Pid}}' $1) 2>/dev/null
+	echo $pid
+}
 
-		# Sleep a bit to let training spawn all workers
-		sleep 120 && echo "Slept 120s, collecting PIDs/TIDs again and ending time_alignment trace"
-		# Capture PIDs/TIDs again now that workload should be in steady state
-		docker top $container_name -efT > ${output_dir}/pids_$(date +'%m%d%H%M%S').out
+attachStraceToPid() {
+	strace -T -tt -f -p $1 -e 'trace=!ioctl,clock_gettime,clock_nanosleep,sched_yield,nanosleep,sched_getaffinity,sched_setaffinity,futex,set_robust_list,poll,epoll_wait,brk' -o ${output_dir}/strace.out &
+}
 
-		# Kill the time alignment trace early, 2min should be plenty
-		kill $trace_time_align_pid
+launchAsExploration() {
+	ps aux -T > ${output_dir}/pids_$(date +'%m%d%H%M%S').out
+	echo "Hit Ctrl-C to stop... the time alignment trace will stop in 60s"
+	read -t 60 -r -d '' _ </dev/tty
+	echo "60s have passed, killing time_alignment trace"
+	kill $trace_time_align_pid
+	echo "Hit Ctrl-C to stop..."
+	read -r -d '' _ </dev/tty
+	echo "Ctrl-C received. Stopping trace"
+}
 
-		echo "Now waiting until training completion"
+performStrace() {
+	if $use_strace; then
+		attachStraceToPid $root_pid_launched_workload
+	fi
+	sleep 120 && echo "Slept 120s, collecting PIDs/TIDs again and ending time_alignment trace"
+	for ((i=0; i<$num_jobs; i++)); do
+		docker top "${container_name}-${i}" -efT >> ${output_dir}/pids_$(date +'%m%d%H%M%S').out
+	done
+	
+	kill $trace_time_align_pid
+	echo "Now waiting until training completion"
+}
 
-		while kill -0 "$root_pid"; do
+waitAllTrainingFinished() {
+	for pid in "${jobs_pids[@]}"; do
+		while kill -0 "$pid"; do
 			sleep 5
 		done
-	else
-		# In exploration mode, we stop the time alignment trace after 60s 
-		# then keep running until Ctrl-C is received
-		ps aux -T > ${output_dir}/pids_$(date +'%m%d%H%M%S').out
-	
-		echo "Hit Ctrl-C to stop... the time alignment trace will stop in 60s"
+	done
+}
 
-		# idle waiting for ctrl-c from user for 60s
-		read -t 60 -r -d '' _ </dev/tty
 
-		echo "60s have passed, killing time_alignment trace"
-		# Kill the time alignment trace early, 2min should be plenty
-		kill $trace_time_align_pid
+usage() { 
+	echo -e "Usage: $0 [OPTIONS] [WORKLOAD ARGS]"
+	echo -e "\nOptions:"
+	echo -e "  -h, --help\t\t\tPrint this message"
+	echo -e "  -w, --workload\t\tName of the workload. Must be one of bert, dlrm, imseg, dlio or explore"
+	echo -e "                  \t\tWith explore, traces are launched without attaching to a particular workload"
+	echo -e "  -l, --launch-script\t\tPath to the workload launch script"
+	echo -e "  -c, --container\t\tName to give the docker container running the workload - defaults to train_{workload}"
+	echo -e "  -n, --num-gpus\t\tNumber of GPUs to launch the workload with, defaults to 1"
```

Comment:
Need to update this for num-jobs -- and specify that it's only for two of the same workload (if i understand correctly, since we launch the traces for only one workload)